src/NeinLinq/SelectorTranslator.cs

```diff
@@ -39,16 +39,26 @@ public static class SelectorTranslator
             var leftInit = left.Body as MemberInitExpression;
             var rightInit = right.Body as MemberInitExpression;
 
-            if (leftInit == null || rightInit == null)
-                throw new NotSupportedException("Only member init expressions are supported yet.");
-            if (leftInit.NewExpression.Arguments.Any() || rightInit.NewExpression.Arguments.Any())
-                throw new NotSupportedException("Only parameterless constructors are supported yet.");
+            var leftNew = left.Body as NewExpression ?? leftInit?.NewExpression;
+            var rightNew = right.Body as NewExpression ?? rightInit?.NewExpression;
+
+            var bindings = Enumerable.Empty<MemberBinding>();
+
+            if (leftNew != null && rightNew != null)
+            {
+                if (leftNew.Arguments.Any() || rightNew.Arguments.Any())
+                    throw new NotSupportedException("Only parameterless constructors are supported yet.");
+
+                if (leftInit != null && rightInit != null)
+                    bindings = leftInit.Bindings.Concat(rightInit.Bindings);
```

Comment:
The whole selector stuff isn't that elegant. However, not needing a new list should be more efficient; a bit. Feel free to push another commit, if that makes sense.